bodhi/models/models.py

```diff
@@ -1112,6 +1106,28 @@ def set_request(self, db, action, username):
             if not self.release.candidate_tag in self.get_tags():
                 self.add_tag(self.release.candidate_tag)
 
+        # If status is pending going to testing request and action is revoke,
+        # set the status to unpushed
+        if self.status is UpdateStatus.pending and self.request is UpdateRequest.testing \
+                and action is UpdateRequest.revoke:
+            self.status = UpdateStatus.unpushed
+            self.revoke()
+            flash_log("%s has been revoked." % self.title)
+            notifications.publish(topic=topic, msg=dict(
+                update=self, agent=username))
+            return
+
+        # If status is testing going to stable request and action is revoke,
+        # keep the status at testing
+        if self.status is UpdateStatus.testing and self.request is UpdateRequest.stable \
+                and action is UpdateRequest.revoke:
+            self.status = UpdateStatus.testing
+            self.revoke()
+            flash_log("%s has been revoked." % self.title)
+            notifications.publish(topic=topic, msg=dict(
+                update=self, agent=username))
+            return
+
```

Comment:
So what happens to a request that is in testing, not pending for anything and being revoked?